SPEC.md

```diff
@@ -70,34 +70,97 @@ A tag consists of the following parts, which, if present, must occur in the list
 
   - **Style class list**
 
-    List of class names, where each class name is prefixed by a `.`. A class name must be a valid *identifier* with the following additionally permitted characters: `-_`. For HTML templates, multiple class names will be merged into a single "class" attribute (classes separated by space characters).
+    List of class names, where each class name is prefixed by a `.`. A class name must be a valid *identifier* with the following additionally permitted characters: `-_`. For HTML templates, multiple class names will be merged into a single "class" attribute (classes separated by space characters). If the tag name has been omitted but at least one class name is present, the tag name will be `div` for HTML templates.
 
   - **Attribute list**
     
-    List of attributes of the form `(att1=value, att2)`. Attributes can have new lines but the children must still be indented as usual. An attribute must be a valid *identifier*. The value part can take any of the following forms:
+    List of attributes of the form `(att1=value, att2)`. Attributes can have new lines but the children must still be indented as usual. An attribute name must be a valid *identifier*. The value part can take any of the following forms:
 
-    - Valid D expression
-    - String literal with double-quotes, which may contain *interpolations*
-    - String literal with single-quotes, which may contain *interpolations*
-    - No value part - equivalent to a boolean value of `true`
+    - a valid D expression
+
+      ```d
+      a(href=user.url) Me
+      // Generates (user = {url: "/bob"})
+      <a href="/bob">Me</a>
+      ```
+
+      which gets compiled into the executable and can use runtime values passed via the render function, see [Embedded D code](#embedded-d-code).
+
+    - a string literal with double-quotes `"` or single-quotes `'`, which may contain *interpolations*
+
+      ```d
+      img(src="/images/avatar_#{picture.id}.png")
+      // Generates (picture = {id: 4})
+      <img src="/images/avatar_4.png"/>
+      ```
+
+    - a boolean value or no value part. It looks like a normal HTML5 shortened attribute but will generate valid XHTML attributes.
+
+      ```d
+      button(enabled)
+      button(enabled=false)
+      // Generates
+      <button enabled="enabled"></button>
+      <button></button>
+      ```
 
   - **Whitespace-removal directives**
 
     - A single `<` will instruct the generator not to emit additional white space within the generated HTML element.
+    ```html
+    div
+    foo>
+      a bar
+    // Generates
+    <div></div><foo>
+      <a>bar</a>
+    </foo>
```

Comment:
Uh I think I just took those from unittests, but I can change it if I accidentally took the wrong ones